File: 프로그래머스/0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.java
class Solution {
    public int solution(int n) {
        //이거 n이랑 6의 최소공배수 찾는 문제인데
        //lmc,gcd 알고리즘 외워라
        
        int a = n;
        int b = 6;
        int gcd = 0;
        //gcd부터 구하자
        
        while(b !=0){
            int temp = a%b;
            a = b;
            b = temp;
        }
        gcd = a;
        
        //lmc 구하자
        int answer = ((n*6)/gcd)/6;
        return answer;
    }
}

File: 프로그래머스/0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.js
function solution(n) {
    //n과 6의 최소공배수 ,단 6만빼고
    let answer = 0;
    if (n===6){
        answer = 1;
    }
    else
        for(let i=1; i<=600; i++){
            if((6*i)%n===0){
                answer = ((6*i)/6);
                break;
            }
        }
    return answer;
}